bigtable-hbase-1.x-parent/bigtable-hbase-1.x-mapreduce/src/main/java/com/google/cloud/bigtable/mapreduce/validation/BigtableSyncMapper.java

```diff
@@ -0,0 +1,187 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.mapreduce.validation;
+
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.SOURCE_BT_APP_PROFILE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.SOURCE_BT_INSTANCE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.SOURCE_BT_PROJECTID_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.TARGET_BT_APP_PROFILE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.TARGET_BT_INSTANCE_CONF_KEY;
+import static com.google.cloud.bigtable.mapreduce.validation.BigtableSyncTableJob.TARGET_BT_PROJECTID_CONF_KEY;
+
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+import com.google.cloud.bigtable.hbase.BigtableOptionsFactory;
+import java.io.IOException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.client.ClusterConnection;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.hbase.mapreduce.BigtableSyncTableAccessor;
+import org.apache.hadoop.hbase.mapreduce.SyncTable.SyncMapper;
+import org.apache.hadoop.mapreduce.Mapper;
+
+/** Bigtable SyncMapper */
+public class BigtableSyncMapper extends SyncMapper {
+
+  private static final Log LOG = LogFactory.getLog(BigtableSyncMapper.class);
+
+  @Override
+  public void setup(
+      Mapper<ImmutableBytesWritable, Result, ImmutableBytesWritable, Mutation>.Context context)
+      throws IOException {
+
+    // modify configuration to establish valid connection
+    Configuration conf = context.getConfiguration();
+    conf.unset(ClusterConnection.HBASE_CLIENT_CONNECTION_IMPL);
+
+    // since both source/target require establishing a connection, set the source as the target or
+    // target as source if not set and connecting with bigtable
+    String targetZkClusterConf = BigtableSyncTableAccessor.getTargetZkClusterConf(conf);
+    String sourceZkClusterConf = BigtableSyncTableAccessor.getSourceZkClusterConf(conf);
+    if (null == targetZkClusterConf && null != sourceZkClusterConf) {
+      BigtableSyncTableAccessor.setTargetZkClusterConf(conf, sourceZkClusterConf);
+      LOG.info(
+          "target conn temporarily set as source for initialization only: " + sourceZkClusterConf);
+    }
+
+    if (null == sourceZkClusterConf && null != targetZkClusterConf) {
+      BigtableSyncTableAccessor.setSourceZkClusterConf(conf, targetZkClusterConf);
+      LOG.info(
+          "source conn temporarily set as target for initialization only: " + targetZkClusterConf);
```

Comment:
please avoid abbreviating things in logs, it'll make it easier for users to understand

Reply:
thanks for the suggestion, updated.